lms/djangoapps/courseware/grades.py

```diff
@@ -499,10 +559,13 @@ def _progress_summary(student, course, course_structure=None):
         course_structure = get_course_blocks(student, course.location)
     if not len(course_structure):
         return None
-    scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
-
-    with outer_atomic():
-        scores_client = ScoresClient.create_for_locations(course.id, student.id, scorable_locations)
+    if scores_client is None:
+        if grading_context_result is None:
+            scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
+        else:  # TODO: these two list are equivalent, yeah?
```

Comment:
I think you have to be careful when refactoring this code.  We don't want to change the resulting behavior.

So:
When creating a common list to be shared by both, yes, use the larger set.
But then for the code that needs the smaller set, make sure it does the additional filtering as a subsequent localized step.
What is the expected performance gain from this work?  I want to see if it's worth the risk at this time - given that we'll be changing the grading code substantially over the next few sprints.